frontend/src/components/lineage/edge/CurvedEdge.tsx

```diff
@@ -0,0 +1,171 @@
+  curveOffset?: number;
+}
+
+type CurvedEdgeInnerProps = Omit<CurvedEdgeProps, 'element'> & { element: Edge };
+
+const CurvedEdgeInner: React.FunctionComponent<CurvedEdgeInnerProps> = observer(
+  ({
+    element,
+    edgeStyle,
+    animationDuration,
+    startTerminalType = EdgeTerminalType.none,
+    startTerminalClass,
+    startTerminalStatus,
+    startTerminalSize = 14,
+    endTerminalType = EdgeTerminalType.directional,
+    endTerminalClass,
+    endTerminalStatus,
+    endTerminalSize = 6,
+    children,
+    className,
+    selected,
+    onSelect,
+    curveOffset = 80,
+  }) => {
+    const startPoint = element.getStartPoint();
+    const endPoint = element.getEndPoint();
+
+    const { shouldRender, isPositioning } = getEdgeVisibilityConfig(element);
+    if (!shouldRender) {
+      return null;
+    }
+
+    // Hide terminal pointers for positioning edges
+    const effectiveStartTerminalType = isPositioning ? EdgeTerminalType.none : startTerminalType;
+    const effectiveEndTerminalType = isPositioning ? EdgeTerminalType.none : endTerminalType;
+
+    const { isConnectedToSelection } = useEdgeHighlighting(element.getId(), selected);
+
+    const { groupClassName, linkClassName, edgeAnimationDuration } = getEdgeStyleConfig({
+      edgeStyle,
+      animationDuration,
+      className,
+      selected,
+      endTerminalStatus,
+      isConnectedToSelection,
+      elementEdgeStyle: element.getEdgeStyle(),
+    });
+
+    const bendpoints = element.getBendpoints();
+
+    const d = createCurvedPath(startPoint, endPoint, bendpoints, curveOffset);
+
+    const backgroundPath = calculateBackgroundPath(
+      startPoint,
+      endPoint,
+      bendpoints,
+      effectiveStartTerminalType,
+      effectiveEndTerminalType,
+      startTerminalSize,
+      endTerminalSize,
+      curveOffset,
+      getConnectorStartPoint,
+    );
+
+    return (
+      <Layer>
+        <g data-test-id="edge-handler" className={groupClassName} onClick={onSelect}>
+          <path
+            className={css(styles.topologyEdgeBackground)}
+            d={backgroundPath}
+            style={getEdgeBackgroundStyles(isPositioning)}
+          />
+          <path
+            className={linkClassName}
+            d={d}
+            style={getEdgeHighlightStyles(
+              isConnectedToSelection,
+              isPositioning,
+              edgeAnimationDuration,
+            )}
+          />
+          <DefaultConnectorTerminal
+            className={startTerminalClass}
+            isTarget={false}
+            edge={element}
+            size={startTerminalSize}
+            terminalType={effectiveStartTerminalType}
+            status={startTerminalStatus}
+          />
+          <DefaultConnectorTerminal
+            className={endTerminalClass}
+            isTarget
+            edge={element}
+            size={endTerminalSize}
+            terminalType={effectiveEndTerminalType}
+            status={endTerminalStatus}
+          />
```

Comment:
You probably want to use a modified version of the `DefaultConnectorTerminal` for the end terminal so that the arrow is always straight into the node rather than based off the edge's incoming angle. 

Alternatively, you could adjust your edge drawing algorithm to straighten out at the end before connecting to the node.

<img width="251" height="565" alt="image" src="https://github.com/user-attachments/assets/ad4ecd92-9cb7-46fd-9c77-77d3f952d7b9" />

Reply:
<img width="971" height="413" alt="Screenshot 2025-08-29 at 1 11 55 PM" src="https://github.com/user-attachments/assets/8e4b447a-a0ad-44e5-b374-e58f8a65491e" />

Improve this